### PR TITLE
feat: implement importing Kubernetes images from a specified cache

### LINKED ISSUE #3896: Image cache provided in install phase
## Feature Request
There is a need to have fully offline install capabilities where all necessary k8s images are provided offline.

### Description
In first boot if a installation cache archive is existing Talos shall unpack and import into containerd to avoid any access to internet.

### PR DESCRIPTION
**Pull request by apariinaz**
When there is a need for complete offline installation of Talos, required Kubernetes images, such as etcd and api server, can be loaded and imported to their containerd namespaces (k8s.io or system) from a specified image-archive cache defined in the node configuration file.

Fixes #3896

Signed-off-by: Parinaz Avaznejad <parinaz.avaznejad@nokia.com>

## Acceptance

Please use the following checklist:

- [x] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5063)

### CODE REVIEW

File: website/content/docs/v0.15/Reference/configuration.md [hunk 1]
```diff
@@ -1475,6 +1475,32 @@ inlineManifests:
...
+
+Settings for Kubernetes images caches.
+
+
+
+Examples:
+
+
+``` yaml
+imageCaches:
```
Comment (frezbo) [inline website/content/docs/v0.15/Reference/configuration.md:1496]: It'd be nice to add to release notes at `hack/release.toml`

**Comment by smira** [comment]
thanks for the PR, some feedback after we did a quick design review:

* with [system extensions](https://github.com/talos-systems/extensions) security of the image cache could be improved so that it resides in the read-only root filesystem vs. the rw `/var`
* also we could probably hardcode the image cache location in the root filesystem getting rid of the configuration option
* we need to discuss the copyright issue
* if the machine config value is required, it should be under `.machine`, not `.cluster`, as this is machine-local configuration

we're happy to talk further about the design of this feature

**Comment by apariinaz** [comment]
> thanks for the PR, some feedback after we did a quick design review:
> 
>     * with [system extensions](https://github.com/talos-systems/extensions) security of the image cache could be improved so that it resides in the read-only root filesystem vs. the rw `/var`
> 
>     * also we could probably hardcode the image cache location in the root filesystem getting rid of the configuration option
> 
>     * we need to discuss the copyright issue
> 
>     * if the machine config value is required, it should be under `.machine`, not `.cluster`, as this is machine-local configuration
> 
> 
> we're happy to talk further about the design of this feature

Thanks for the comments. yes, it would be great if we can discuss mentioned matters. I will get in touch on Slack.

**Comment by apariinaz** [comment]
The feature can be implemented with extension services, so the pull request can be closed.

**Pull request closed without merging**